### PR TITLE
Simple function to clear the DO bit from an OPT resource record.

### PR DESCRIPTION
**Pull request by shane-kerr**

### CODE REVIEW

**Comment by miekg** [comment]
Also see #406. Probably an oversight (from me, not to give SetDo a parameter).

I think btw we can get away with this, but redefining SetDo() as
```
func SetDo(do bool...) {}
```
checking for the number of args, and if it's 1, use that value, otherwise to the current thing.

**Comment by shane-kerr** [comment]
Okay I'll update the branch accordingly - I am a Go n00b so didn't really consider the hand-crated optional parameter rout. :wink:

**Comment by shane-kerr** [comment]
Okay, this is now changed to support an optional argument.

Unfortunately there does not appear to be an easy way in Go to require that there be only one argument. You can return an error, but anyone who accidentally passes 2+ parameters probably is not checking errors carefully. I guess it could call `panic()` in that case?

File: edns_test.go [hunk 1]
```diff
@@ -15,6 +15,18 @@ func TestOPTTtl(t *testing.T) {
...
 	}
+	e.SetDo(true)
+	if !e.Do() {
+		t.Errorf("DO bit should still be non-zero")
+	}
+	e.SetDo(false)
+	if e.Do() {
+		t.Errorf("DO bit should be zero")
+	}
+	e.SetDo(false)
```
Comment (miekg) [inline edns_test.go]: does e.SetDo() also still work?

File: edns.go [hunk 2]
```diff
@@ -128,8 +128,17 @@ func (rr *OPT) Do() bool {
 }
 
 // SetDo sets the DO (DNSSEC OK) bit.
-func (rr *OPT) SetDo() {
-	rr.Hdr.Ttl |= _DO
+func (rr *OPT) SetDo(do ...bool) {
+	// if we pass an argument, set the DO bit to that value
```
Comment (miekg) [inline edns.go]: Put this comment in the exported doc string above the function. Also tell we will silently ignore more than 1 argument.

**Comment by miekg** [comment]
No, panic() in a lib is usually a bad idea.

File: edns_test.go (hunk 1 above)
Comment (shane-kerr) [inline edns_test.go]: Actually this was tested, but there was a missing test or two. I made a comment explaining comprehensive tests for this, and annotated each call to explain which case is covered. Hopefully that makes it clear (and hopefully I didn't mess it up).

File: edns.go (hunk 2 above)
Comment (shane-kerr) [inline edns.go]: Okay, done.

**Comment by shane-kerr** [comment]
The last commit addresses the review comments - thanks!

**Comment by miekg** [comment]
Nice. Thanks.

**Pull request merged**